xarray/backends/zarr.py

```diff
@@ -163,24 +164,36 @@ def _determine_zarr_chunks(enc_chunks, var_chunks, ndim, name):
                 continue
             for dchunk in dchunks[:-1]:
                 if dchunk % zchunk:
-                    raise NotImplementedError(
+                    base_error = (
                         f"Specified zarr chunks encoding['chunks']={enc_chunks_tuple!r} for "
                         f"variable named {name!r} would overlap multiple dask chunks {var_chunks!r}. "
-                        "This is not implemented in xarray yet. "
-                        "Consider either rechunking using `chunk()` or instead deleting "
-                        "or modifying `encoding['chunks']`."
+                        f"Writing this array in parallel with dask could lead to corrupted data. "
                     )
+                    if safe_chunks:
+                        raise ValueError(
```

Comment:
really minor comment: Shouldn't this still be `NotImplementedError` since we could technically support this by implementing locks?

Reply:
It used to be, but I changed it! Do we ever plan to implement locks?